src/functions.date.js

```diff
@@ -108,28 +108,43 @@ for (let unit in s) {
 	}, {multiValued: true});
 }
 
-_.duration = $.extend(function($this, ms) {
+_.duration = $.extend(function($this, ms, terms = 1) {
 	if (arguments.length === 1) {
 		[ms, $this] = [$this, null];
 	}
-
-	var count = ms || 0;
-	var unit = "ms";
-
-	for (let nextUnit in s) {
-		var nextCount = _.msTo(nextUnit, ms);
-
-		if (nextCount === 0) {
-			break;
+	
+	let unitTime = ms || 0;
+	let timeLeft = ms || 0;
+	
+	if (ms === 0) {
+		terms = 1;
+	}
+	
+	let timeArray = [];
+	
+	while (timeArray.length < terms && (timeLeft > 0 || terms === 1)) {
+		let unit = "ms";
+		
+		for (let nextUnit in s) {
+			let count = _.msTo(nextUnit, timeLeft);
+			
+			if (count === 0) {
+				break;
+			}
+			
+			unitTime = count;
+			unit = nextUnit;
```

Comment:
I may have misunderstood your algorithm, but it seems to me that this inner loop only runs once if `_.msTo()` returns 0, otherwise it keeps going and overwriting previous results? Why is that?

In general, this is at the point of complexity that it would have benefitted from a few comments, at least to explain what this loop does, and what the conditional below (the rest of the codebase is not necessarily a shining example of sufficient comments usage either, I know…).

Reply:
Good point; I'll add in a clarifying comment. This loop only runs once if _.msTo() returns 0 b/c nextUnit starts with the lowest-valued term apart from ms: seconds. So if a seconds conversion of the timeLeft is 0, then we will stick with the current unit of ms (which is why we break).